application/src/main/java/net/corda/v5/application/crypto/DigestService.java

```diff
@@ -40,4 +40,14 @@ public interface DigestService {
      */
     @Suspendable
     int digestLength(@NotNull DigestAlgorithmName digestName);
-}
+
+
+    /**
+     * Creates a [SecureHash].
+     * <p>
+     * This function does not validate the length of the created digest.
+     */
+
+    @Suspendable
+    SecureHash parse(@NotNull String str);
```

Comment:
I don't like the function naming - it doesn't parse a `DigestService`, so it should say what it's parsing.
```suggestion
    SecureHash parseSecureHash(@NotNull String str);
```